### DISCUSSION TITLE
Postgres connection

### DISCUSSION THREAD

**Discussion by wildsonc** [apache/superset#18275] [Q&A / Help]
I'm trying to link the superset to postgres, but when I try to import a dashboard it crashes. I also can't save charts.

#### Screenshots

![image](https://user-images.githubusercontent.com/72408418/108253301-644ff100-7138-11eb-9b24-5107813dde3d.png)
![image](https://user-images.githubusercontent.com/72408418/108253425-86497380-7138-11eb-9048-c46e97e5f4c9.png)

#### How to reproduce the bug

1. Go to Import Dashboard
2. Click on Upload
3. See error.

### Environment

- superset version: 1.0.1
- python version: Python 3.8.5
- postgres version: psql (PostgreSQL) 12.6 (Ubuntu 12.6-0ubuntu0.20.04.1)
- Instaled from Scratch

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ x] I have reproduced the issue with at least the latest released version of superset.
- [ x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Log

ERROR:superset.app:Exception on /superset/import_dashboards [POST]
Traceback (most recent call last):
  File "/root/superset/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/root/superset/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/root/superset/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/root/superset/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/root/superset/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/root/superset/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/root/superset/lib/python3.8/site-packages/flask_appbuilder/security/decorators.py", line 109, in wraps
    return f(self, *args, **kwargs)
  File "/root/superset/lib/python3.8/site-packages/superset/utils/log.py", line 164, in wrapper
    value = f(*args, **kwargs)
  File "/root/superset/lib/python3.8/site-packages/superset/views/core.py", line 660, in import_dashboards
    databases = db.session.query(Database).all()
  File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3373, in all
    return list(self)
  File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3535, in __iter__
    return self._execute_and_instances(context)
  File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3556, in _execute_and_instances
    conn = self._get_bind_args(
  File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3571, in _get_bind_args
    return fn(
  File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3550, in _connection_from_session
    conn = self.session.connection(**kw)
  File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 1142, in connection
    return self._connection_for_bind(
  File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 1150, in _connection_for_bind
    return self.transaction._connection_for_bind(
  File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 409, in _connection_for_bind
    self._assert_active()
  File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 289, in _assert_active
    raise sa_exc.InvalidRequestError(
sqlalchemy.exc.InvalidRequestError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: 'ascii' codec can't encode character '\xe1' in position 7: ordinal not in range(128) (Background on this error at: http://sqlalche.me/e/13/7s2a)
INFO:werkzeug:170.83.134.254 - - [17/Feb/2021 18:50:15] "POST /superset/import_dashboards HTTP/1.1" 500 -
INFO:werkzeug:170.83.134.254 - - [17/Feb/2021 18:54:59] "GET /superset/import_dashboards HTTP/1.1" 200 -

**Comment by willbarrett** [comment]
@betodealmeida you're the most familiar with import - thoughts?

**Comment by SanDoz145** [comment]
@betodealmeida If you have the most familiarity with the import function - please help. We are trying to export dashboard from Ver 0.34 to 1.0 and facing the same "unknown error" issue.

**Comment by DrVajiha** [comment]
> I'm trying to link the superset to postgres, but when I try to import a dashboard it crashes. I also can't save charts.
> #### Screenshots
> 
> ![image](https://user-images.githubusercontent.com/72408418/108253301-644ff100-7138-11eb-9b24-5107813dde3d.png) ![image](https://user-images.githubusercontent.com/72408418/108253425-86497380-7138-11eb-9048-c46e97e5f4c9.png)
> #### How to reproduce the bug
> 
>     1. Go to Import Dashboard
> 
>     2. Click on Upload
> 
>     3. See error.
> 
> 
> ### Environment
> 
>     * superset version: 1.0.1
> 
>     * python version: Python 3.8.5
> 
>     * postgres version: psql (PostgreSQL) 12.6 (Ubuntu 12.6-0ubuntu0.20.04.1)
> 
>     * Instaled from Scratch
> 
> 
> ### Checklist
> 
> Make sure to follow these steps before submitting your issue - thank you!
> 
>     * [x ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
> 
>     * [ x] I have reproduced the issue with at least the latest released version of superset.
> 
>     * [ x] I have checked the issue tracker for the same issue and I haven't found one similar.
> 
> 
> ### Log
> 
> ERROR:superset.app:Exception on /superset/import_dashboards [POST] Traceback (most recent call last): File "/root/superset/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app response = self.full_dispatch_request() File "/root/superset/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request rv = self.handle_user_exception(e) File "/root/superset/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception reraise(exc_type, exc_value, tb) File "/root/superset/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise raise value File "/root/superset/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request rv = self.dispatch_request() File "/root/superset/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request return self.view_functions[rule.endpoint](**req.view_args) File "/root/superset/lib/python3.8/site-packages/flask_appbuilder/security/decorators.py", line 109, in wraps return f(self, *args, **kwargs) File "/root/superset/lib/python3.8/site-packages/superset/utils/log.py", line 164, in wrapper value = f(*args, **kwargs) File "/root/superset/lib/python3.8/site-packages/superset/views/core.py", line 660, in import_dashboards databases = db.session.query(Database).all() File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3373, in all return list(self) File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3535, in **iter** return self._execute_and_instances(context) File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3556, in _execute_and_instances conn = self._get_bind_args( File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3571, in _get_bind_args return fn( File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3550, in _connection_from_session conn = self.session.connection(**kw) File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 1142, in connection return self._connection_for_bind( File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 1150, in _connection_for_bind return self.transaction._connection_for_bind( File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 409, in _connection_for_bind self._assert_active() File "/root/superset/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 289, in _assert_active raise sa_exc.InvalidRequestError( sqlalchemy.exc.InvalidRequestError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: 'ascii' codec can't encode character '\xe1' in position 7: ordinal not in range(128) (Background on this error at: http://sqlalche.me/e/13/7s2a) INFO:werkzeug:170.83.134.254 - - [17/Feb/2021 18:50:15] "POST /superset/import_dashboards HTTP/1.1" 500 - INFO:werkzeug:170.83.134.254 - - [17/Feb/2021 18:54:59] "GET /superset/import_dashboards HTTP/1.1" 200 -

Hi, I am also facing the same issue. I'm trying to export dashboard from server to another server of same version 1.5, but showing error: Integrity error. 
@wildsonc Have you fix this issue. If anybody fixed this issue. Kindly guide me to import dashboard.

**Comment by DrVajiha** [comment]
Hi, I am also facing the same issue. I'm trying to export dashboard from server to another server of same version 1.5, but showing error: Integrity error. If anybody fixed this issue. Kindly guide me to import dashboard.

ERROR: (while importing Dashboard)

sqlalchemy.exc.InvalidRequestError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (sqlite3.IntegrityError) UNIQUE constraint failed: tables.table_name
[SQL: INSERT INTO tables (uuid, created_on, changed_on, description, default_endpoint, is_featured, filter_select_enabled, "offset", cache_timeout, params, perm, schema_perm, is_managed_externally, external_url, table_name, main_dttm_col, database_id, fetch_values_predicate, schema, sql, is_sqllab_view, template_params, extra, created_by_fk, changed_by_fk) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)]
[parameters: (<memory at 0x7f3b7c882640>, '2022-11-18 11:22:47.890965', '2022-11-18 11:22:47.890971', None, None, 0, 0, 0, None, '{"remote_id": 19, "database_name": "PostgreSQL", "import_time": 1668750767}', None, None, 0, None, 'storefrequency', None, '7', None, None, None, 0, None, None, 1, 1)]
(Background on this error at: http://sqlalche.me/e/13/gkpj) (Background on this error at: http://sqlalche.me/e/13/7s2a)
2022-11-18 11:22:47,893:INFO:werkzeug:127.0.0.1 - - [18/Nov/2022 11:22:47] "POST /superset/import_dashboards/ HTTP/1.1" 500 -
